codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/protocols/ServerHttpBoundProtocolGenerator.kt

```diff
@@ -722,7 +723,15 @@ class ServerHttpBoundProtocolTraitImplGenerator(
             inputShape.serverBuilderSymbol(codegenContext),
         )
         Attribute.AllowUnusedVariables.render(this)
-        rust("let (parts, body) = request.into_parts();")
+        rustTemplate(
+            """
+            let #{RequestParts} { uri, headers, body, .. } = #{Request}::try_from(request)?.into_parts();
+            """,
+            *preludeScope,
```

Comment:
Do we need `preludeScope` here? We should add it to `codegenScope` instead.

Better yet, I want `rustTemplate` to have the prelude scope baked in, but that's a separate PR.

Reply:
https://github.com/smithy-lang/smithy-rs/issues/3167

Reply:
yes I want this too!

Reply:
I also want this, but I'll keep it out of this PR. Looks like I can remove the prelude from that rustTemplate call.